core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -328,25 +318,95 @@ private void xContentHeader(XContentBuilder builder, String key, List<String> va
     }
 
     /**
-     * Static toXContent helper method that also renders non {@link org.elasticsearch.ElasticsearchException} instances as XContent.
+     * Renders additional per exception information into the XContent
      */
-    public static void toXContent(XContentBuilder builder, Params params, Throwable ex) throws IOException {
-        ex = ExceptionsHelper.unwrapCause(ex);
-        if (ex instanceof ElasticsearchException) {
-            ((ElasticsearchException) ex).toXContent(builder, params);
+    protected void innerToXContent(XContentBuilder builder, Params params) throws IOException {
+    }
+
+    /**
+     * Indicates if the cause of the {@link ElasticsearchException} must be skipped (true) or rendered
+     * as part of the XContent (false).
+     *
+     * @param params parameters used to build the XContent
+     * @return a boolean indicating if the cause must be skipped (true) or rendered (false)
+     */
+    protected boolean skipCauseInXContent(Params params) {
+        return params.paramAsBoolean(REST_EXCEPTION_SKIP_CAUSE, REST_EXCEPTION_SKIP_CAUSE_DEFAULT);
+    }
+
+    /**
+     * Static toXContent helper method that also renders non {@link org.elasticsearch.ElasticsearchException} instances as XContent,
+     * delegating the rendering to {@link ElasticsearchException#toXContent(XContentBuilder, Params)}.
+     */
+    public static void toXContent(XContentBuilder builder, Params params, Throwable t) throws IOException {
+        t = ExceptionsHelper.unwrapCause(t);
+
+        ElasticsearchException exception;
+        if (t instanceof ElasticsearchException) {
+            exception = (ElasticsearchException) t;
         } else {
-            builder.field(TYPE, getExceptionName(ex));
-            builder.field(REASON, ex.getMessage());
-            if (ex.getCause() != null) {
-                builder.field(CAUSED_BY);
+            final Throwable throwable = t;
+            exception = new ElasticsearchException(throwable.getMessage(), throwable.getCause()) {
```

Comment:
maybe instead of wrapping the exception we should have a static method that takes as an argument all the little pieces that get printed out, then that method could be called here and in toXContent?